# Puzzle

https://thefiddler.substack.com/p/when-will-you-cross-your-path

**This Week’s Fiddler**

Anita the ant is going for a walk in the sand, leaving a trail as she goes. First, she walks 1 inch in a straight line. Then she rotates counterclockwise by an angle 𝝋, after which she walks another 2 inches. She rotates counterclockwise an angle 𝝋 again, after which she walks 3 inches. She keeps doing this over and over again, rotating counterclockwise an angle 𝝋 and then walking 1 inch farther than she did in the previous segment.

At some point during her journey, she crosses over her initial 1-inch segment. By “cross over,” I am including the two end points of that first segment.

Anita realizes that 𝝋 was the smallest possible angle such that she crossed over her 1-inch segment. (Among the ants, she’s known for her mathematical prowess.)

How long was the segment along which she first crossed over the 1-inch segment? Your answer should be a whole number of inches.

**This Week’s Extra Credit**

It’s time for you to check Anita’s work. What was the measure of angle 𝝋?

Remember, this was the smallest possible angle for each turn such that she crossed over her 1-inch segment at some later point.

# Solution

Modelling the steps and the positions using complex numbers seems like a good choice because it has a tidy way to include the rotations.

Let $ T = e^{i\varphi} $ be the turning factor. 

Assuming Anita starts at the origin, her position after n steps will be:

$ P_n = 1 + 2T + 3T^2 + 4T^3 + ... + nT^{n-1} $

This is a well known structure ( https://en.wikipedia.org/wiki/Arithmetico-geometric_sequence ), and we can simplify it in a standard way.

$ P_n = (1 + T + ... + T^{n-1}) + (T + .. T^{n-1}) ... + (T^{n-1}) $

$ \displaystyle P_n = \frac{T^n - 1}{T-1} + \frac{T^n - T}{T-1} + ... + \frac{T^n - T^{n-1}}{T-1} $

$ \displaystyle P_n = \frac{nT^n}{T-1} - \frac{1 + T + ... + T^{n-1}}{T-1} $

$ \displaystyle P_n = \frac{nT^n}{T-1} - \frac{T^n - 1}{(T-1)^2} $

$ \displaystyle P_n = \frac{nT^n(T-1) - (T^n - 1)}{(T-1)^2} $

$ \displaystyle P_n = \frac{nT^{n+1} - (n+1)T^n + 1}{(T-1)^2} $

If we plug in n=1 and simplify, we get $ \displaystyle P_n = \frac{T^2 - 2T + 1}{(T-1)^2} = 1 $, which is correct.

If we plug in n=2 and simplify, we get $ \displaystyle P_n = \frac{2T^3 - 3T^2 + 1}{(T-1)^2} = \frac{(2T + 1)(T^2 - 2T + 1}{(T-1)^2} = 1 + 2T $, which is correct.

So, the formula looks good.

Not sure how to use this to get an analytical answer at this point, but it at least gives us a numerically superior way of calculating positions (compared to summing large sequences), esp if we choose to use $ T^n = e^{in\varphi} $ (which I won't, but still ... :) )

So, now I will switch to desmos or python to try to figure out what the answer should be before we try to prove it more rigorously (if possible).

# Desmos

It was relatively easy to create a graph in desmos that shows the path of Anita the Ant for any given angle $\varphi$ and number of steps $N$. It's very fun to play with the sliders.

https://www.desmos.com/calculator/snfcneiuqe

Sliding the value of $\varphi$ around I could see that until $\varphi ~= 2.4$ the path simply diverges without intersecting itself at any point, and something around 2.4 is the time it intersects itself.

Looking at the path at this point shows that the solution to the Fiddler is 4, i.e. the intersection happens at the 4th step.

To calculate the value of $\varphi$ more precisely, I just zoomed into the intersection point and adjusted the value of $\varphi$ by typing in values numerically (i.e. not using the slider). With repeated iterations of this, it is clear that the desired value of $\varphi$ is $\varphi ~= 2.4188584058 \  rad$. 

Here's a snapshot of that process : https://www.desmos.com/calculator/0tp4n68ixc

# Exact Value

Looking at the desmos diagram from the matching condition, we can see a triangle. 

The first side is the step of length 2, the second side is the step of length 3, and the third side is some fraction of of the step of length 4. 

Since the angles between the sides are constant, we can infer that this is an isoceles triangle and the third side length is 2. 

So, this triangle has sides 2, 3, 2. Let both the angles around the base (side length 3) be $\theta = \pi - \varphi$ .

Applying some basic trignometry, we can calculate that:

$\theta = arccos(\frac{1.5}{2}) = arccos(\frac{3}{4})$

And so:

$\varphi = \pi - arccos(\frac{3}{4})$

In [1]:
from math import pi, acos, degrees
theta_rad = acos(3/4)
theta_deg = degrees(theta_rad)

phi_rad = pi - theta_rad
phi_deg = degrees(phi_rad)
print(f"theta (radians): {theta_rad}, theta (degrees): {theta_deg}")
print(f"phi (radians): {phi_rad}, phi (degrees): {phi_deg}")

theta (radians): 0.7227342478134157, theta (degrees): 41.40962210927086
phi (radians): 2.4188584057763776, phi (degrees): 138.59037789072914


In very good agreement with the empirical result from desmos.

# Conclusion

The Fiddler Answer (length of crossing segment) is 4 inches.

The Extra Credit Answer (turning angle $\varphi$) is 2.4188584057763776 radians = 138.59037789072914 degrees.